lib/Twig/Node/Expression/GetAttr.php

```diff
@@ -23,6 +23,25 @@ public function __construct(Twig_Node_Expression $node, Twig_Node_Expression $at
 
     public function compile(Twig_Compiler $compiler)
     {
+        // optimize array calls
+        if (
+            (!$compiler->getEnvironment()->isStrictVariables() || $this->getAttribute('ignore_strict_check'))
+            && !$this->getAttribute('is_defined_test')
+            && Twig_Template::ARRAY_CALL === $this->getAttribute('type')
+        ) {
+            $compiler
+                ->raw('(is_array(')
```

Comment:
shouldn't it be `is_array($node) || $node instanceof ArrayAccess` ?

Reply:
indeed, that's even better